src/firecracker/src/api_server_adapter.rs

```diff
@@ -149,25 +149,35 @@ pub(crate) fn run_with_api(
         .name("fc_api".to_owned())
         .spawn(move || {
             match ApiServer::new(to_vmm, from_vmm, to_vmm_event_fd).bind_and_run(
-                api_bind_path,
+                &api_bind_path,
                 process_time_reporter,
                 &api_seccomp_filter,
                 api_payload_limit,
                 socket_ready_sender,
             ) {
                 Ok(_) => (),
-                Err(api_server::Error::Io(inner)) => match inner.kind() {
-                    std::io::ErrorKind::AddrInUse => panic!(
-                        "Failed to open the API socket: {:?}",
-                        api_server::Error::Io(inner)
-                    ),
-                    _ => panic!(
-                        "Failed to communicate with the API socket: {:?}",
-                        api_server::Error::Io(inner)
-                    ),
-                },
-                Err(eventfd_err @ api_server::Error::Eventfd(_)) => {
-                    panic!("Failed to open the API socket: {:?}", eventfd_err)
+                Err(api_server::Error::ServerCreation(ServerError::IOError(inner))) => {
```

Comment:
It would be better if the thread returned a result rather than calling `std::process::exit`. This can then be propagated up through `run_with_api` and eventually back to main, calling `std::process::exit` is non-ideal (https://rust-lang.github.io/rust-clippy/master/#exit).

Reply:
This is something we can do as an overall improvement for the Firecracker code (make the std::process::exit get propagated to main) if possible (the std::process::exit was there before this PR) -> you can open an issue for this.

For the moment, this PR actually solves a bug in which some code would never get called and the error message triggered when the API socket cannot be opened is not descriptive of the problem.